SlackPirate.py

```diff
@@ -535,29 +535,48 @@ def file_cleanup(input_file, output_info: OutputInformation):
+    parser.add_argument('--no-s3-scan', action='store_true', help='skips searching for s3 references in messages')
+    parser.add_argument('--no-credential-scan', action='store_true',
+                        help='skips searching for messages referencing credentials')
+    parser.add_argument('--no-aws-key-scan', action='store_true', help='skips search for aws keys in messages')
+    parser.add_argument('--no-private-key-scan', action='store_true', help='skips search for private keys in messages')
+    parser.add_argument('--no-link-scan', action='store_true', help='skips searching for interesting links')
+    parser.add_argument('--no-file-download', action='store_true', help='skips downloading of files from the workspace')
     parser.add_argument('--version', action='version',
                         version='SlackPirate.py v0.2. Developed by Mikail Tunç (@emtunc) with contributions from '
                                 'the amazing community! https://github.com/emtunc/SlackPirate/graphs/contributors')
     args = parser.parse_args()
 
-    if args.cookie is None and args.token is None:
+    if args.cookie is None and args.token is None:  # Must provide one or the other
         print(termcolor.colored("No arguments passed. Run SlackPirate.py --help ", "white", "on_red"))
         exit()
-    elif args.cookie and args.token:
+    elif args.cookie and args.token:  # May not provide both
         print(termcolor.colored("You cannot use both --cookie and --token flags at the same time", "white", "on_red"))
         exit()
-
-    if args.cookie:
+    elif args.cookie:  # Providing a cookie leads to a shorter execution path
         display_cookie_tokens(cookie=dict(d=args.cookie))
-    else:
-        provided_token = args.token
-        collected_output_info = check_token_validity(token=provided_token)
-        print_interesting_information(output_info=collected_output_info)
-        dump_team_access_logs(token=provided_token, output_info=collected_output_info)
-        dump_user_list(token=provided_token, output_info=collected_output_info)
-        find_s3(token=provided_token, output_info=collected_output_info)
-        find_credentials(token=provided_token, output_info=collected_output_info)
-        find_aws_keys(token=provided_token, output_info=collected_output_info)
-        find_private_keys(token=provided_token, output_info=collected_output_info)
-        find_interesting_links(token=provided_token, output_info=collected_output_info)
-        download_interesting_files(token=provided_token, output_info=collected_output_info)
+        exit()
+    # Baseline behavior
+    provided_token = args.token
+    collected_output_info = check_token_validity(token=provided_token)
+    print_interesting_information(output_info=collected_output_info)
+
+    # Possible scans to run along with flags that disable them
+    flags_and_scans = [
+        ('no_team_access_log', dump_team_access_logs),
```

Comment:
Note that in order to continue this pattern, the scan function must maintain the interface of having `token` and `output_info` as parameters. If additional parameters are needed, then currying or partial functions may be used. It might be worth explicitly defining the scan function signature.